### ISSUE TITLE
Problem with submitting SAN domains using New-AcmeCertificate -AlternativeIdentifierRefs

### ISSUE THREAD

**Issue by centur** [ebekker/ACMESharp#141]
Hi,
I tried to follow the quick start guide and generate a certificate with multiple SANs in it, but for some reason `New-ACMECertificate` gives me an error:

```
New-ACMECertificate : The given key was not present in the dictionary.
At line:1 char:1
+ New-ACMECertificate <skipped> -AlternativeIdentifierRefs  ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [New-ACMECertificate], KeyNotFoundException
    + FullyQualifiedErrorId : System.Collections.Generic.KeyNotFoundException,ACMESharp.POSH.NewCertificate
```

Here is the sequence of commands I run to get to the problem.

- `$sans` is an array of domains I want in SAN field
- All these domains were previously validated with dns-01 challenge and I can generate certs of all of them individually on the same machine.
- I've tried to run `New-ACMECertificate domain.in.sans.list -AlternativeIdentifierRefs $sans -Alias "existing alias" -Generate` for the same domain  in $sans 
- I've also tried to run `New-ACMECertificate domain.not.in.sans.list -AlternativeIdentifierRefs $sans -Alias "existing alias" -Generate` for the domain which is not in $sans list

Both were unsuccessful. Below is the screen with output of the commands

![le-san-issue](https://cloud.githubusercontent.com/assets/720822/15662226/ed1213b6-2731-11e6-9272-5acf86c5e0e1.png)

What am I doing wrong ?

**Comment by RyanShelf** [comment]
I'm having this same issue, good to know I'm not alone here.

**Comment by bseddon** [comment]
Nice comment, all the relevant information provided clearly.

I believe the problem is that you have used the domain names in the $sans array not the aliases.  The command will try to resolve the aliases contained in $sans and find, for example, 'devbuilds.drawboard.com' and fail.  It will fail because this is not an alias.  The alias for this domain is 'devbuilds.drawboard' (without the '.com').

It looks similar so when you are trying to find the problem it's not going stand out.  I know this from my own experience.  As you have with your May-17 and May-18 aliases I always include a date so I know for sure when I'm using aliases.

**Comment by centur** [comment]
Ok, so AlternativeIdentifierRef is an Alias, Thank you, will try this and report about the results.
Is it possible to write some small terminology glossary for the module, as it's a bit confusing sometimes....

**Comment by centur** [comment]
Yep, it works. For anyone who stumbles over this - take a note that because `AlternativeIdentifierRefs` expects an array - you need to wrap your other valid domain alias into array construction like this `@('drawboard')`

Example:
```
New-ACMECertificate api.drawboard.com.2016-May-23_09-28 -AlternativeIdentifierRefs @('drawboard') -Alias api.drawboard.com.2016-May-23_09-28-alias -Generate
```

**Comment by ebekker** [comment]
Related to #104 -- I'll enhance the CLI interface to support single entries or arrays properly.
